pkg/shell/runner.go

```diff
@@ -65,14 +74,30 @@ func (r *shellRunner) executeCmd(obfuscate func([]string), cmdExe string, cmdArg
 		obfuscate(displayValues)
 	}
 	logrus.Infof("Running command %s %s", cmdExe, strings.Join(displayValues, " "))
-	return r.execute(cmd)
+	var output io.Writer
+	var buffer bytes.Buffer
+	if readOutputs {
+		output = &buffer
+	}
+	err = r.execute(cmd, output)
+	if err != nil {
+		return
+	}
+	if readOutputs {
+		outputs = buffer.String()
+	}
+	return
 }
 
-func (r *shellRunner) execute(cmd *exec.Cmd) error {
+func (r *shellRunner) execute(cmd *exec.Cmd, output io.Writer) error {
 	cmd.Env = append(cmd.Env, os.Environ()...)
 	cmd.Env = append(cmd.Env, r.context.Export()...)
 	cmd.Stdin = os.Stdin
-	cmd.Stdout = os.Stdout
+	if output != nil {
+		cmd.Stdout = io.MultiWriter(os.Stdout, output)
```

Comment:
Will it also write image digest query cmd output to os.stdout and eventually go to user's build log? It seems not necessary.

Reply:
Not sure about this one. I always show command and its outputs and I think we should still show the outputs even if the output is captured by the builder and used.